### DISCUSSION TITLE
Goda Pressures Calculations

### DISCUSSION THREAD

**Discussion by fgarcia0128** [coastal-hazards-system/StormSim-Library#29] [General]
StormSim Goda Pressures have the following things to sort out:

- [ ] Logic to handle Rc<0 entries
- [ ] Decide if NaN entries should be set to 0 or NaN (Impacts plots)
- [x] Logic to use h when berm = 0
- [ ] Add impulsive pressures ? (Already in S2G function, just needs integrating)
replacing image with one from CEM.  This is the definition of the parameters we want to use.
![image](https://user-images.githubusercontent.com/18451506/227320257-9a43d710-549e-4f40-8b46-2c18bdd197bf.png)
In the image, hs is the input depth at the location that wave conditions are defined.  So that would normally be the save point depth, chs_sp_depth.  d is the top of mound and h' is the bottom of wall, hw is the total height of wall, Bm is width of berm crest seaward of wall.  These last four params need to be defined by the user so we need 4 additional inputs under a new category of Goda pressure inputs.  The remainder of the parameters are computed.  We can add the sloping face modification of Table VI-5-56 as well.  This requires that the user input the structure slope angle. but this is not a priority.
@jmelby

**Comment by KHodgens** [comment]
"hs is the input depth at the location that wave conditions are defined. So that would normally be the save point depth, chs_sp_depth."

Is this statement only true for flat bottoms?

The info that follows above Goda image seems to require a lot more considerations for hb, Hdesign for the various alpha parameters, plus whether or not the structure is landward or seaward of breaking.  Are we calculating all of those?
![image](https://github.com/Coastal-Hazards-System/StormSim-Library/assets/113923029/a0105d11-c4a7-4626-a266-fccc1ec0faf9)
